lib/charms/tls_certificates_interface/v3/tls_certificates.py

```diff
@@ -687,6 +691,68 @@ def _get_certificate_expiry_time(certificate: str) -> Optional[datetime]:
         return None
 
 
+def _get_certificate_validity_start_time(certificate: str) -> Optional[datetime]:
+    """Extract expiry time from a certificate string.
+
+    Args:
+        certificate (str): x509 certificate as a string
+
+    Returns:
+        Optional[datetime]: Time when this certificate is valid from or None
+    """
+    try:
+        certificate_object = x509.load_pem_x509_certificate(data=certificate.encode())
+        return certificate_object.not_valid_before_utc
+    except ValueError:
+        logger.warning("Could not load certificate.")
```

Comment:
Is this warning accurate? Could it be possible that we can load the certificate but it doesn't contain the `not_valid_before_utc` field?

Reply:
I'm not sure there can be an x509 cert without such field. But I will need to validate that from the RFC before being certain.

Reply:
It can be [without the after one](https://github.com/canonical/tls-certificates-interface/issues/148), at least.

Reply:
I think that error is because of the version of the `cryptography` lib being used, but I'm not sure which tls provider was used there.
After looking into [RFC](https://datatracker.ietf.org/doc/html/rfc5280.html#section-4.1.2.5) it seems like validity should be defined, and when we load a valid certificate using `x509.load_pem_x509_certificate` (having `cryptography>42.0.0`) then we will be able to get the "not before" and "not after" dates using both the deprecated `not_valid_before/not_valid_after` or the `not_valid_before_utc/not_valid_after_utc`.

Reply:
- I added a constraint for cryptography version in the docstring.
- I changed the behavior to skip certificates that don't have validity time set.

Please let me know what you think and whether we want this change or not (I'd say yes). And let me know if you have any other suggestions on how to do this.